Example/src/screens/HeaderOptions.tsx

```diff
@@ -0,0 +1,181 @@
+  const [backButtonVisible, setBackButtonVisible] = useState(true);
+  const [headerShown, setHeaderShown] = useState(true);
+  const [headerLargeTitle, setHeaderLargeTitle] = useState(false);
+  const [headerItem, setHeaderItem] = useState('right');
+  const [headerBackTitle, setHeaderBackTitle] = useState('Back');
+  const [headerHideShadow, setHeaderHideShadow] = useState(false);
+  const [headerTranslucent, setHeaderTranslucent] = useState(false);
+
+  const square = (props: {tintColor?: string}) => (
+    <Square {...props} color="green" size={20} />
+  );
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle,
+      headerBackTitleVisible: backButtonVisible, // iOS
+      headerHideBackButton: !backButtonVisible, // android
+      headerShown,
+      headerRight: headerItem === 'right' ? square : undefined,
+      headerCenter: headerItem === 'center' ? square : undefined,
+      headerLeft: headerItem === 'left' ? square : undefined,
+      headerLargeTitle, // iOS
+      headerBackTitle, // iOS
+      headerHideShadow,
+      headerTranslucent,
+    });
+  }, [
+    navigation,
+    headerTitle,
+    backButtonVisible,
+    headerItem,
+    headerShown,
+    headerLargeTitle,
+    headerBackTitle,
+    headerHideShadow,
+    headerTranslucent,
+  ]);
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <SettingsInput
+        label="Header title"
+        value={headerTitle}
+        onValueChange={setHeaderTitle}
+      />
+      <SettingsSwitch
+        label="Back button visible"
+        value={backButtonVisible}
+        onValueChange={setBackButtonVisible}
+      />
+      <SettingsSwitch
+        label="Header shown"
+        value={headerShown}
+        onValueChange={setHeaderShown}
+      />
+      <SettingsSwitch
+        label="Header hide shadow"
+        value={headerHideShadow}
+        onValueChange={setHeaderHideShadow}
+      />
+      <SettingsSwitch
+        label="Header translucent"
+        value={headerTranslucent}
+        onValueChange={setHeaderTranslucent}
+      />
+      <SettingsPicker
+        label="Header item"
+        value={headerItem}
+        onValueChange={setHeaderItem}
+        items={['left', 'center', 'right']}
+      />
+      {Platform.OS === 'ios' ? (
+        <>
+          <Text style={styles.heading}>iOS only</Text>
+          <SettingsSwitch
+            label="Header large title"
+            value={headerLargeTitle}
+            onValueChange={setHeaderLargeTitle}
+          />
+          <SettingsInput
+            label="Header back title"
+            value={headerBackTitle}
+            onValueChange={setHeaderBackTitle}
+          />
+        </>
+      ) : null}
+      <Button title="Go back" onPress={() => navigation.goBack()} />
+    </SafeAreaView>
+  );
+};
+
+const Stack = createNativeStackNavigator<StackParamList>();
+
+const App = (): JSX.Element => (
+  <Stack.Navigator
+    screenOptions={{
+      statusBarStyle: 'dark',
```

Comment:
same

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/827/commits/85e8e17b756652507b71956a30319822422c5358